OpenImis.ePayment/Data/ImisBasePayment.cs

```diff
@@ -821,6 +821,117 @@ public void CancelPayment(int payment_id)
             }
         }
 
+        public object GetPaymentToReconciliate(object payment)
+        {
+            object result = null;
+            double paidAmount = Convert.ToDouble(payment.GetType().GetProperty("paidAmount").GetValue(payment));
+            SqlParameter[] parameters = {
+                new SqlParameter("@Id", payment.GetType().GetProperty("billId").GetValue(payment)),
+                new SqlParameter("@paidAmount", paidAmount),
+            };
+           
+            var sSQL = @"SELECT PaymentId, ExpectedAmount, ReceivedAmount, PaymentStatus FROM tblPayment WHERE PaymentId=@Id And PaymentStatus<=5 And ExpectedAmount=@paidAmount And ValidityTo is Null";
```

Comment:
Can you use some Declaration in the SQL or add the PaymentStatus as variable and use the constants defined based on a later comment in this PR?

Reply:
Ok. Sure! Clear for me what do you want here.